### PR TITLE
Separate block list from blockchain API struct

### PR DESCRIPTION
**Pull request by prince-chrismc**
## Description

Fixes pyrsia/pyrsia#529 

This is to make it easier to add the encoding/decoding traits that need to be support for Aleph which counts towards #322

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).
- [x] I've requested a review  from "pyrsia/collaborators"

### Code Contributions

- [x] I've built the code `cargo build --workspace`.
- [x] I've run the unit tests `cargo test --workspace`.

### CODE REVIEW

File: src/blockchain/src/blockchain.rs [hunk 1]
```diff
@@ -191,18 +127,20 @@ mod tests {
...
+            TransactionType::Create,
+            local_id,
+            data.as_bytes().to_vec(),
             &keypair,
         );
         transactions.push(transaction);
-        chain.add_block(new_block(
+        chain.add_block(Block::new(
+            chain.blocks()[0].header.hash(),
+            chain.blocks()[0].header.ordinal,
```
Comment (prince-chrismc) [inline src/blockchain/src/blockchain.rs:138]: Applying feedback from https://github.com/pyrsia/pyrsia/pull/517/files/693e1e7b227cac24f141ade09dfa9c0fabbefa8b#r837860004

```suggestion
            chain.blocks()[0].header.ordinal+1,
```

File: src/blockchain/src/structures/chain.rs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+use serde::{Deserialize, Serialize};
+
+use super::block::Block;
+
+#[derive(Serialize, Deserialize, Debug, Default, Clone)]
+pub struct Chain {
```
Comment (prince-chrismc) [inline src/blockchain/src/structures/chain.rs:22]: Being able to have default traits on the "binary data" was to be able to do #535 with relative easy

**Review by fishseabowl — APPROVED** [review]
LGTM

**Review by sbtaylor15 — APPROVED** [review]
no changes required.

**Review by erwin1 — APPROVED** [review]

**Pull request merged**